### ISSUE TITLE
OpenImageDenoise 2.0: Need update old one to new

### ISSUE THREAD

**Issue by odil24** [LuxCoreRender/BlendLuxCore#884]
@neo2068 any idea?

**Comment by neo2068** [comment]
There is a documentation of the needed changes for update 1.x to 2: [link](https://www.openimagedenoise.org/documentation.html)

**Comment by odil24** [comment]
> There is a documentation of the needed changes for update 1.x to 2: [link](https://www.openimagedenoise.org/documentation.html)

Yeah, I see. Need to refactor many things which is really time waster

**Comment by howetuft** [comment]
Please note the wheels are already built against OIDN **2.3.0**

**Comment by howetuft** [comment]
I advocate a thorough overhaul of the way dependencies are managed in LuxCore, by relying on a **dependency manager** (I recommend Conan), rather than continuing to patch the current system of static embedded dependencies.

**Comment by neo2068** [comment]
The imagepipeline of LuxCore uses the old 1.x OIDN library and . The code needs to be changed to work with 2.x OIDN and support execution on GPU. Did you adapt the code for the wheels build? The use of a dependency management system sounds like a good idea.

**Comment by howetuft** [comment]
No, I did not adapt the code. I just targeted OIDN 2.3.1 to build against (and it worked!) but, at this stage, I did nothing to trigger the GPU. This is another step...
The wheel builder is here, for more precisions: https://github.com/howetuft/LuxCoreWheel and I'm available for any question.

**Comment by odil24** [comment]
@neo2068 @howetuft From where are we need to start on implementing GPU side code for working latest OIDN?

**Comment by howetuft** [comment]
My first priority is to embed `oidnDenoise` into the wheels, to make them fully efficient for BlendLuxCore (see Discord exchanges)
My second priority will be to be sure to embed the GPU libs into the wheels, so that it _might_ be possible to use the GPU. At the moment, for simplification, I focused on CPU, except on MacOS.

Afterwards, effectively trigger the GPU in deoising will require 3 prerequisites I may lack:
- technical ability to make the imagepipeline evolve in the core codebase (LuxCoreRender)
- commit rights on core codebase
- time to code that upgrade

**Comment by odil24** [comment]
> My first priority is to embed `oidnDenoise` into the wheels, to make them fully efficient for BlendLuxCore (see Discord exchanges)
> My second priority will be to be sure to embed the GPU libs into the wheels, so that it _might_ be possible to use the GPU. At the moment, for simplification, I focused on CPU, except on MacOS.
> 
> Afterwards, effectively trigger the GPU in denoising will require 3 prerequisites I may lack:
> - technical ability to make the imagepipeline evolve in the core codebase (LuxCoreRender)
> - commit rights on core codebase
> - time to code that upgrade

Yeah, it is seems more tricky thing than just fix issues and add some commitments into the BlendLuxCore addon. Anyways i am not familiar with your relationship and thats why I will pass it on my side.
